README.md

```diff
@@ -163,23 +163,19 @@ logically-connected sync/async code execution.
 namespace AsyncContext {
   class Variable<T> {
     constructor(options: AsyncVariableOptions<T>);
-
     get name(): string;
-
     run<R>(value: T, fn: (...args: any[])=> R, ...args: any[]): R;
-
     get(): T | undefined;
   }
-
   interface AsyncVariableOptions<T> {
     name?: string;
     defaultValue?: T;
   }
 
   class Snapshot {
     constructor();
-
     run<R>(fn: (...args: any[]) => R, ...args: any[]): R;
+    wrap<T, R>(fn: (this: T, ...args: any[]) => R): (this: T, ...args: any[]) => R;
```

Comment:
Why would someone want to be defensive against a non-conformant AsyncContext implementation? I would imagine this more like,

`callback = typeof AsyncContext === "object" ? AsyncContext.Snapshot.wrap(callback) : callback`

Reply:
I'm OK with either position, but I definitely don't think a second `?.` should be encouraged. I also kinda imagine using a utility function that would isolate this from most consumer code:

```js
const wrap = (cb) => AsyncContext?.Snapshot.wrap(cb) ?? cb;


class Foo {
  test() {
    const cb = wrap(() => …);
  }
}

Reply:
I don't feel strongly.

Reply:
`AsyncContext.wrap` would be ambiguous since a function can also be wrapped with `AsyncContext.Variable.prototype.run`. I would find `AsyncContext.Snapshot.wrap` be more explicit that a function is wrapped with a snapshot.

Reply:
> `AsyncContext.wrap` would be ambiguous since a function can also be wrapped with `AsyncContext.Variable.prototype.run`

Re: https://github.com/tc39/proposal-async-context/issues/25

> I would find `AsyncContext.Snapshot.wrap` be more explicit that a function is wrapped with a snapshot.

Let's keep as is then.